src/Structures/Tree.php

```diff
@@ -45,7 +45,7 @@ public function tree($tree = null)
     {
         return $this->fluentlyGetOrSet('tree')
             ->getter(function ($tree) {
-                $key = "structure-{$this->handle()}-{$this->locale()}-".md5(json_encode($tree));
+                $key = "structure-{$this->handle()}-{$this->locale()}-".md5(json_encode($this->tree));
```

Comment:
The `$tree` would be `$this->tree`. Are you sure this made a difference?

Reply:
Yeah but I've no real idea why... when you get() could `$tree` be `null` and it was always returning the same md5()?

Reply:
For fuller explanation the issue was when you added something to the tree, it was still getting the blink cached version so it appeared that the tree hadn't updated. Its not as big an issue as the rest of the PR and I'm happy to back it out.

Reply:
Ah I see, maybe it was how in some places we call `$this->tree()` but others just `$this->tree`. Maybe. Thanks though.

Reply:
No problem, done!